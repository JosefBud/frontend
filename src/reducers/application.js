import { buildActions } from 'utils';

export const types = buildActions('application', [
  'INIT_APP',
  'LOGIN_USER',
  'POP_TOAST',
  'ADD_TOAST',
  'REMOVE_TOAST',
  'HIDE_TOAST',
  'REQUEST_TOKEN',
  'REQUEST_TOKEN_SUCCESS',
  'REQUEST_TOKEN_FAILURE',
  'REQUEST_CURRENT_USER',
  'REQUEST_CURRENT_USER_SUCCESS',
  'REQUEST_CURRENT_USER_FAILURE',
  'LOGIN_USER_SUCCESS',
  'LOGIN_USER_FAILURE',
  'LOGOUT_USER',
  'LOGOUT_USER_SUCCESS',
  'LOGOUT_USER_FAILURE',
  'REGISTER_USER',
  'REGISTER_USER_SUCCESS',
  'REGISTER_USER_FAILURE',
  'REQUEST_USERS',
  'REQUEST_USERS_SUCCESS',
  'REQUEST_USERS_FAILURE',
  'REQUEST_MIGRATIONS',
  'REQUEST_MIGRATIONS_SUCCESS',
  'REQUEST_MIGRATIONS_FAILURE',
  'REQUEST_ROLES',
  'REQUEST_ROLES_SUCCESS',
  'REQUEST_ROLES_FAILURE',
  'REQUEST_FLAVORS',
  'REQUEST_FLAVORS_SUCCESS',
  'REQUEST_FLAVORS_FAILURE'
]);

const initApp = () => ({
  type: types.INIT_APP
});

const loginUser = (emailAddress, password) => ({
  type: types.LOGIN_USER,
  emailAddress,
  password
});

const popToast = toast => ({
  type: types.POP_TOAST,
  toast
});

const addToast = toast => ({
  type: types.ADD_TOAST,
  toast
});

const removeToast = id => ({
  type: types.REMOVE_TOAST,
  id
});

const hideToast = id => ({
  type: types.HIDE_TOAST,
  id
});

const requestToken = (emailAddress, password) => ({
  type: types.REQUEST_TOKEN,
  emailAddress,
  password
});

const requestTokenSuccess = ({ token, expiration }) => ({
  type: types.REQUEST_TOKEN_SUCCESS,
  expiration,
  token
});

const requestTokenFailure = error => ({
  type: types.REQUEST_TOKEN_FAILURE,
  error
});

const requestCurrentUser = () => ({
  type: types.REQUEST_CURRENT_USER
});

const requestCurrentUserSuccess = user => ({
  type: types.REQUEST_CURRENT_USER_SUCCESS,
  user
});

const requestCurrentUserFailure = error => ({
  type: types.REQUEST_CURRENT_USER_FAILURE,
  error
});

const loginUserSuccess = () => ({
  type: types.LOGIN_USER_SUCCESS
});

const loginUserFailure = error => ({
  type: types.LOGIN_USER_FAILURE,
  error
});

const logoutUser = () => ({
  type: types.LOGOUT_USER
});

const logoutUserSuccess = () => ({
  type: types.LOGOUT_USER_SUCCESS
});

const logoutUserFailure = error => ({
  type: types.LOGOUT_USER_FAILURE,
  error
});

const registerUser = details => ({
  type: types.REGISTER_USER,
  details
});

const registerUserSuccess = () => ({
  type: types.REGISTER_USER_SUCCESS
});

const registerUserFailure = error => ({
  type: types.REGISTER_USER_FAILURE,
  error
});

const requestUsers = () => ({
  type: types.REQUEST_USERS
});

const requestUsersSuccess = users => ({
  type: types.REQUEST_USERS_SUCCESS,
  users
});

const requestUsersFailure = error => ({
  type: types.REQUEST_USERS_FAILURE,
  error
});

const requestMigrations = () => ({
  type: types.REQUEST_MIGRATIONS
});

const requestMigrationsSuccess = migrations => ({
  type: types.REQUEST_MIGRATIONS_SUCCESS,
  migrations
});

const requestMigrationsFailure = error => ({
  type: types.REQUEST_MIGRATIONS_FAILURE,
  error
});

const requestRoles = () => ({
  type: types.REQUEST_ROLES
});

const requestRolesSuccess = roles => ({
  type: types.REQUEST_ROLES_SUCCESS,
  roles
});

const requestRolesFailure = error => ({
  type: types.REQUEST_ROLES_FAILURE,
  error
});

const requestFlavors = () => ({
  type: types.REQUEST_FLAVORS
});

const requestFlavorsSuccess = flavors => ({
  type: types.REQUEST_FLAVORS_SUCCESS,
  flavors
});

const requestFlavorsFailure = error => ({
  type: types.REQUEST_FLAVORS_FAILURE,
  error
});

export const actions = {
  initApp,
  loginUser,
  popToast,
  addToast,
  removeToast,
  hideToast,
  requestToken,
  requestTokenSuccess,
  requestTokenFailure,
  requestCurrentUser,
  requestCurrentUserSuccess,
  requestCurrentUserFailure,
  loginUserSuccess,
  loginUserFailure,
  logoutUser,
  logoutUserSuccess,
  logoutUserFailure,
  registerUser,
  registerUserSuccess,
  registerUserFailure,
  requestUsers,
  requestUsersSuccess,
  requestUsersFailure,
  requestMigrations,
  requestMigrationsSuccess,
  requestMigrationsFailure,
  requestRoles,
  requestRolesSuccess,
  requestRolesFailure,
  requestFlavors,
  requestFlavorsSuccess,
  requestFlavorsFailure
};

export const initialState = {
  loggingIn: false,
  loggingOut: false,
  user: null,
  error: null,
  authorization: {
    accessToken: null,
    refreshToken: null,
    expiration: null
  },
  registration: {
    registering: false,
    complete: false,
    error: null
  },
  toasts: [],
  users: [],
  migrations: [],
  roles: [],
  flavors: []
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.ADD_TOAST:
      return {
        ...state,
        toasts: [...state.toasts, action.toast]
      };
    case types.REMOVE_TOAST:
      return {
        ...state,
        toasts: state.toasts.filter(toast => toast.id !== action.id)
      };
    case types.HIDE_TOAST: {
      const originalToast = state.toasts.find(toast => toast.id === action.id);

      if (!originalToast) {
        return state;
      }

      const newToast = {
        ...originalToast,
        show: false
      };
      const filteredToasts = state.toasts.filter(
        toast => toast.id !== action.id
      );

      return {
        ...state,
        toasts: [...filteredToasts, newToast]
      };
    }
    case types.LOGIN_USER:
      return {
        ...state,
        loggingIn: true
      };
    case types.LOGOUT_USER:
      return {
        ...state,
        loggingOut: true
      };
    case types.REGISTER_USER:
      return {
        ...state,
        registration: {
          ...state.registration,
          registering: true,
          details: action.details
        }
      };
    case types.REQUEST_TOKEN_SUCCESS:
      return {
        ...state,
        authorization: {
          ...state.authorization,
          accessToken: action.token,
          expiration: action.expiration
        }
      };
    case types.REQUEST_TOKEN_FAILURE:
      return {
        ...state,
        authorization: {
          ...state.authorization,
          error: action.error
        }
      };
    case types.REQUEST_CURRENT_USER_SUCCESS:
      return {
        ...state,
        user: action.user
      };
    case types.LOGOUT_USER_SUCCESS:
      return {
        ...state,
        loggingOut: false,
        user: null,
        authorization: initialState.authorization
      };
    case types.LOGIN_USER_SUCCESS:
      return {
        ...state,
        loggingIn: false
      };
    case types.LOGIN_USER_FAILURE:
    case types.LOGOUT_USER_FAILURE:
      return {
        ...state,
        loggingIn: false,
        loggingOut: false,
        error: action.error
      };
    case types.REGISTER_USER_SUCCESS:
      return {
        ...state,
        registration: {
          registering: false,
          complete: true,
          error: null
        }
      };
    case types.REGISTER_USER_FAILURE:
      return {
        ...state,
        registration: {
          registering: false,
          complete: true,
          error: action.error
        }
      };
    case types.REQUEST_USERS_SUCCESS:
      return {
        ...state,
        users: action.users
      };
    case types.REQUEST_USERS_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.REQUEST_MIGRATIONS_SUCCESS:
      return {
        ...state,
        migrations: action.migrations
      };
    case types.REQUEST_MIGRATIONS_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.REQUEST_ROLES_SUCCESS:
      return {
        ...state,
        roles: action.roles
      };
    case types.REQUEST_ROLES_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.REQUEST_FLAVORS_SUCCESS:
      return {
        ...state,
        flavors: action.flavors
      };
    case types.REQUEST_FLAVORS_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
