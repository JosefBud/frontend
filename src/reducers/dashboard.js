import { buildActions } from 'utils';

export const types = buildActions('dashboard', [
  'REQUEST_MIGRATIONS',
  'REQUEST_MIGRATIONS_SUCCESS',
  'REQUEST_MIGRATIONS_FAILURE'
]);

const requestMigrations = () => ({
  type: types.REQUEST_MIGRATIONS
});

const requestMigrationsSuccess = migrations => ({
  type: types.REQUEST_MIGRATIONS_SUCCESS,
  migrations
});

const requestMigrationsFailure = error => ({
  type: types.REQUEST_MIGRATIONS_FAILURE,
  error
});

export const actions = {
  requestMigrations,
  requestMigrationsSuccess,
  requestMigrationsFailure
};

export const initialState = {
  migrations: []
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_MIGRATIONS_SUCCESS:
      return {
        ...state,
        migrations: action.migrations
      };
    case types.REQUEST_MIGRATIONS_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
