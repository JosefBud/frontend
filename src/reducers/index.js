import { combineReducers } from 'redux';
import {
  reducer as application,
  initialState as applicationState
} from './application';
import {
  reducer as dashboard,
  initialState as dashboardState
} from './dashboard';
import {
  reducer as dataSuppliers,
  initialState as dataSuppliersState
} from './dataSuppliers';
import { reducer as flavors, initialState as flavorsState } from './flavors';
import { reducer as roles, initialState as rolesState } from './roles';
import { reducer as toast, initialState as toastState } from './toast';
import { reducer as users, initialState as usersState } from './users';

export const initialState = {
  application: applicationState,
  dashboard: dashboardState,
  dataSuppliers: dataSuppliersState,
  flavors: flavorsState,
  roles: rolesState,
  toast: toastState,
  users: usersState
};

export default combineReducers({
  application,
  dashboard,
  dataSuppliers,
  flavors,
  roles,
  toast,
  users
});
