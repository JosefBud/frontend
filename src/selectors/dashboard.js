import { createSelector } from 'reselect';

export const getDashboard = state => state.dashboard;

export const getDashboardComponent = createSelector(
  getDashboard,
  dashboard => dashboard.dashboard
);

export const getMigrations = createSelector(
  getDashboard,
  dashboard => dashboard.migrations
);
