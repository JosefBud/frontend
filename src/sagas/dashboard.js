// import dayjs from 'dayjs';
// import nanoid from 'nanoid';
import {
  all,
  call,
  put,
  takeLatest
  // delay,
  // take,
  // takeEvery,
  // select
} from 'redux-saga/effects';

import request from 'utils/request';
import { actions, types } from 'reducers/dashboard';

// snake_cased variables here come from RFC 6749
/* eslint-disable camelcase */
function* requestMigrationsWorker() {
  try {
    const endpoint = {
      url: '/data/version',
      method: 'GET'
    };
    const result = yield call(request.execute, { endpoint });

    // update migrations in state or throw an error
    if (result.success) {
      const {
        response: { data }
      } = result;

      yield put(actions.requestMigrationsSuccess(data));
    } else if (result.error) {
      throw result.error;
    } else {
      throw new Error('Failed to get migrations!');
    }
  } catch (error) {
    const { message } = error;

    yield put(actions.requestMigrationsFailure(error));
    yield put(
      actions.popToast({
        title: 'Error',
        icon: 'times-circle',
        message
      })
    );
  }
}

function* requestMigrationsWatcher() {
  yield takeLatest(types.REQUEST_MIGRATIONS, requestMigrationsWorker);
}

export const workers = {
  requestMigrationsWorker
};

export const watchers = {
  requestMigrationsWatcher
};

export default function* saga() {
  yield all(Object.values(watchers).map(watcher => watcher()));
}
