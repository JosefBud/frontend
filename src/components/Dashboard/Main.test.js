import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { initialState } from 'reducers/dashboard';
import ConnectedMain from './Main';
import { withMemoryRouter } from 'utils';

describe('Dashboard <Main />', () => {
  const mockStore = configureStore();
  const store = mockStore({ dashboard: initialState });
  const RoutedMain = withMemoryRouter(ConnectedMain);

  it('renders correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <RoutedMain />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
