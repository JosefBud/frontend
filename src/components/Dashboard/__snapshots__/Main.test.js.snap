// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard <Main /> renders Flavors correctly 1`] = `
<div
  className="card"
  style={Object {}}
>
  <div
    className="card-header"
  >
    Ingredients &gt; Flavors
  </div>
  <div
    className="card-body"
  >
    <div
      className="table-responsive"
    >
      <table
        className="table table-sm table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              ID
            </th>
            <th>
              Vendor
            </th>
            <th>
              Name
            </th>
            <th>
              Slug
            </th>
            <th>
              Density
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td>
              Capella
            </td>
            <td>
              27 Bears
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              2
            </td>
            <td>
              Capella
            </td>
            <td>
              27 Fish
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              3
            </td>
            <td>
              Capella
            </td>
            <td>
              Acai
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              4
            </td>
            <td>
              Capella
            </td>
            <td>
              Amaretto
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              5
            </td>
            <td />
            <td>
              Anise
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              6
            </td>
            <td />
            <td>
              Apple Pie
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              7
            </td>
            <td />
            <td>
              Apple Pie v2
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              8
            </td>
            <td />
            <td>
              Apple Snacks
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              9
            </td>
            <td />
            <td>
              Apricot
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              10
            </td>
            <td />
            <td>
              Banana
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              11
            </td>
            <td />
            <td>
              Banana Split
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              12
            </td>
            <td />
            <td>
              Bavarian Cream
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              13
            </td>
            <td />
            <td>
              Biscuit
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              15
            </td>
            <td />
            <td>
              Blackberry
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              14
            </td>
            <td />
            <td>
              Black Currant
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              17
            </td>
            <td />
            <td>
              Blueberry
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              18
            </td>
            <td />
            <td>
              Blueberry Cinnamon Crumble
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              19
            </td>
            <td />
            <td>
              Blueberry Extra
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              20
            </td>
            <td />
            <td>
              Blueberry Jam
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              21
            </td>
            <td />
            <td>
              Blueberry Pomegranate
            </td>
            <td />
            <td />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Dashboard <Main /> renders Home correctly 1`] = `
Array [
  <div
    className="card-columns"
  >
    <div
      className="card bg-danger text-white"
    >
      <div
        className="card-header"
      >
        Users: 
        2000
      </div>
      <div
        className="card-body"
      >
        <h3>
          Activated Users: 
          1991
        </h3>
        <h3>
          Inactive Users: 
          9
        </h3>
        <h3>
          Active Tokens: 
          300
        </h3>
      </div>
    </div>
    <div
      className="card bg-primary text-white"
    >
      <div
        className="card-body"
      >
        <h2>
          Tags: 
          115
        </h2>
      </div>
    </div>
    <div
      className="card bg-secondary text-white"
    >
      <div
        className="card-body"
      >
        <h2>
          Vendors: 
          98
        </h2>
      </div>
    </div>
    <div
      className="card bg-info text-white"
    >
      <div
        className="card-header"
      >
        Flavors: 
        8617
      </div>
      <div
        className="card-body"
      >
        <h2>
          Tags: 
          90
        </h2>
      </div>
    </div>
    <div
      className="card bg-warning text-white"
    >
      <div
        className="card-header"
      >
        Recipes: 
        42
      </div>
      <div
        className="card-body"
      >
        <h2>
          Tags: 
          55
        </h2>
      </div>
    </div>
  </div>,
  <a
    href="#home"
    onClick={[Function]}
    title="Refresh Stats"
  >
    <i
      className="fa sync"
    />
  </a>,
]
`;

exports[`Dashboard <Main /> renders Migations correctly 1`] = `
<div
  className="card"
  style={Object {}}
>
  <div
    className="card-header"
  >
    Database &gt; Migrations
  </div>
  <div
    className="card-body"
  >
    <div
      className="table-responsive"
    >
      <table
        className="table table-sm table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
              Name
            </th>
            <th>
              Ran
            </th>
            <th>
              MD5
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td>
              initial-tables
            </td>
            <td>
              2019-08-05T04:08:51.887Z
            </td>
            <td>
              21a37c3b009b10c825eeb503316d8455
            </td>
          </tr>
          <tr>
            <td>
              2
            </td>
            <td>
              initial-data
            </td>
            <td>
              2019-08-05T04:08:52.133Z
            </td>
            <td>
              b7e138254b5aa7e7f5a2c4b01a47a4bf
            </td>
          </tr>
          <tr>
            <td>
              3
            </td>
            <td>
              remove-extra
            </td>
            <td>
              2019-08-05T04:08:52.152Z
            </td>
            <td>
              1c21f069a9eba5b6a90c3d7e1000b631
            </td>
          </tr>
          <tr>
            <td>
              4
            </td>
            <td>
              oauth
            </td>
            <td>
              2019-08-05T04:08:52.430Z
            </td>
            <td>
              93bbe1fba0fe3a0199bb64da5f588619
            </td>
          </tr>
          <tr>
            <td>
              5
            </td>
            <td>
              vendor-identifier
            </td>
            <td>
              2019-08-05T04:08:52.661Z
            </td>
            <td>
              1e307264fb5e163c731fd3a853188106
            </td>
          </tr>
          <tr>
            <td>
              6
            </td>
            <td>
              cleanup
            </td>
            <td>
              2019-08-05T04:08:52.683Z
            </td>
            <td>
              3ace3654625342de9408b1a02eab779c
            </td>
          </tr>
          <tr>
            <td>
              7
            </td>
            <td>
              preparations-diluents
            </td>
            <td>
              2019-08-05T04:08:52.873Z
            </td>
            <td>
              9df28acfb709b31aa010551d3ee106c8
            </td>
          </tr>
          <tr>
            <td>
              8
            </td>
            <td>
              user-registration
            </td>
            <td>
              2019-08-05T04:08:53.125Z
            </td>
            <td>
              26e034920dcce1f448b9a4161403268d
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              recipe-vendor
            </td>
            <td>
              2019-08-05T04:08:53.258Z
            </td>
            <td>
              eeee5bb71f86970c746a31a30449bba2
            </td>
          </tr>
          <tr>
            <td>
              10
            </td>
            <td>
              millipercent
            </td>
            <td>
              2019-08-05T04:08:53.501Z
            </td>
            <td>
              5ca462e6fbe8bae86c75552b276fa988
            </td>
          </tr>
          <tr>
            <td>
              11
            </td>
            <td>
              prep-diluents-pk
            </td>
            <td>
              2019-08-05T04:08:53.600Z
            </td>
            <td>
              b329bbec7eeb9af639b0964a21deb7a9
            </td>
          </tr>
          <tr>
            <td>
              12
            </td>
            <td>
              tag-and-version
            </td>
            <td>
              2019-08-11T00:57:10.873Z
            </td>
            <td>
              87647de8630b05c444d158a2cf87d3a3
            </td>
          </tr>
          <tr>
            <td>
              13
            </td>
            <td>
              unique-role-name
            </td>
            <td>
              2019-08-11T00:57:10.999Z
            </td>
            <td>
              c6c7d7aabd456322c7fa38068da1014c
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Dashboard <Main /> renders Role/Add correctly 1`] = `
<div
  className="card"
  style={Object {}}
>
  <div
    className="card-header"
  >
    Roles &gt; Add Role
  </div>
  <div
    className="card-body"
  >
    <i
      className="fa chevron-left"
    />
      
    <a
      href="#roles"
      onClick={[Function]}
    >
      <span>
        Back
      </span>
    </a>
    <form
      className=""
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Role Name
        </label>
        <input
          className="form-control"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="role name"
          type="text"
          value=""
        />
      </div>
      <button
        className="button-animation btn btn-primary"
        disabled={false}
        type="submit"
      >
        <span>
          Save
        </span>
      </button>
    </form>
  </div>
</div>
`;

exports[`Dashboard <Main /> renders Role/Add/User correctly 1`] = `
<div
  className="card"
  style={Object {}}
>
  <div
    className="card-header"
  >
    Roles &gt; Assign test Role
  </div>
  <div
    className="card-body"
  >
    <i
      className="fa chevron-left"
    />
      
    <a
      href="#roles"
      onClick={[Function]}
    >
      Back
    </a>
    <form
      className=""
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Select User to Assign to 
          test
          s Role
        </label>
        <select
          className="form-control"
          name="userId"
          onChange={[Function]}
          value="0"
        >
          <option
            value="0"
          >
            Choose a User
          </option>
          <option
            value={1}
          >
            blah@blah.com
          </option>
          <option
            value={2}
          >
            blah2@blah.com
          </option>
        </select>
      </div>
    </form>
    <br />
  </div>
</div>
`;

exports[`Dashboard <Main /> renders Role/Delete correctly 1`] = `
<div
  className="card"
  style={Object {}}
>
  <div
    className="card-header"
  >
    Roles &gt; Delete Role &gt; test
  </div>
  <div
    className="card-body"
  >
    <i
      className="fa chevron-left"
    />
      
    <a
      href="#roles"
      onClick={[Function]}
    >
      <span>
        Back
      </span>
    </a>
    <form
      className=""
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Delete
        </label>
        <h3>
          Delete Role: 
          test
          ?
        </h3>
      </div>
      <button
        className="button-animation btn btn-primary"
        disabled={false}
        type="submit"
      >
        <span>
          Delete
        </span>
      </button>
    </form>
    <br />
    <div
      className="card border-danger"
    >
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Roles &gt; test
        </div>
        <i
          className="fa chevron-left"
        />
          
        <a
          href="#roles"
          onClick={[Function]}
        >
          Back
        </a>
        <div
          className="fade alert show"
          role="alert"
        >
          No Users Assigned to this Role!
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dashboard <Main /> renders Role/Delete/User correctly 1`] = `
<div
  className="card"
  style={Object {}}
>
  <div
    className="card-header"
  >
    Roles &gt; Unassign Role &gt; test &gt; User ID: 20
  </div>
  <div
    className="card-body"
  >
    <i
      className="fa chevron-left"
    />
      
    <a
      href="#user/roles/20"
      onClick={[Function]}
    >
      <span>
        Back
      </span>
    </a>
    <form
      className=""
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Unassign Role
        </label>
        <h3>
          Unassign Role: 
          test
           from User ID 
          20
          ?
        </h3>
      </div>
      <button
        className="button-animation btn btn-primary"
        disabled={false}
        type="submit"
      >
        <span>
          Remove
        </span>
      </button>
    </form>
  </div>
</div>
`;

exports[`Dashboard <Main /> renders Role/Edit correctly 1`] = `
<div
  className="card"
  style={Object {}}
>
  <div
    className="card-header"
  >
    Roles &gt; Edit Role &gt; test
  </div>
  <div
    className="card-body"
  >
    <i
      className="fa chevron-left"
    />
      
    <a
      href="#roles"
      onClick={[Function]}
    >
      <span>
        Back
      </span>
    </a>
    <form
      className=""
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Role Name
        </label>
        <input
          className="form-control"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="test"
          type="text"
          value=""
        />
      </div>
      <button
        className="button-animation btn btn-primary"
        disabled={false}
        type="submit"
      >
        <span>
          Save
        </span>
      </button>
    </form>
  </div>
</div>
`;

exports[`Dashboard <Main /> renders Role/Users correctly 1`] = `
<div
  className="card"
  style={Object {}}
>
  <div
    className="card-header"
  >
    Roles &gt; test
  </div>
  <div
    className="card-body"
  >
    <i
      className="fa chevron-left"
    />
      
    <a
      href="#roles"
      onClick={[Function]}
    >
      Back
    </a>
    <div
      className="fade alert show"
      role="alert"
    >
      No Users Assigned to this Role!
    </div>
  </div>
</div>
`;

exports[`Dashboard <Main /> renders Roles correctly 1`] = `
<div
  className="card"
  style={Object {}}
>
  <div
    className="card-header"
  >
    Roles
  </div>
  <div
    className="card-body"
  >
    <a
      href="#role/add"
      onClick={[Function]}
    >
      Add Role
    </a>
    <div
      className="table-responsive"
    >
      <table
        className="table table-sm table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              ID
            </th>
            <th>
              Name
            </th>
            <th>
              Options
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td>
              Administrator
            </td>
            <td>
              <a
                href="#role/users/1"
                onClick={[Function]}
              >
                Users
              </a>
                |  
              <a
                href="#role/1/add/user"
                onClick={[Function]}
              >
                Assign
              </a>
            </td>
          </tr>
          <tr>
            <td>
              2
            </td>
            <td>
              User
            </td>
            <td>
              <a
                href="#role/users/2"
                onClick={[Function]}
              >
                Users
              </a>
                |  
              <a
                href="#role/2/add/user"
                onClick={[Function]}
              >
                Assign
              </a>
            </td>
          </tr>
          <tr>
            <td>
              3
            </td>
            <td>
              Tester
            </td>
            <td>
              <a
                href="#role/users/3"
                onClick={[Function]}
              >
                Users
              </a>
                |  
              <a
                href="#role/3/add/user"
                onClick={[Function]}
              >
                Assign
              </a>
                |  
              <a
                href="#role/edit3"
                onClick={[Function]}
              >
                Edit
              </a>
                |  
              <a
                href="#role/delete3"
                onClick={[Function]}
              >
                Delete
              </a>
            </td>
          </tr>
          <tr>
            <td>
              4
            </td>
            <td>
              Moderator
            </td>
            <td>
              <a
                href="#role/users/4"
                onClick={[Function]}
              >
                Users
              </a>
                |  
              <a
                href="#role/4/add/user"
                onClick={[Function]}
              >
                Assign
              </a>
                |  
              <a
                href="#role/edit4"
                onClick={[Function]}
              >
                Edit
              </a>
                |  
              <a
                href="#role/delete4"
                onClick={[Function]}
              >
                Delete
              </a>
            </td>
          </tr>
          <tr>
            <td>
              8
            </td>
            <td>
              Everyone
            </td>
            <td>
              <a
                href="#role/users/8"
                onClick={[Function]}
              >
                Users
              </a>
                |  
              <a
                href="#role/8/add/user"
                onClick={[Function]}
              >
                Assign
              </a>
                |  
              <a
                href="#role/edit8"
                onClick={[Function]}
              >
                Edit
              </a>
                |  
              <a
                href="#role/delete8"
                onClick={[Function]}
              >
                Delete
              </a>
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              Myself
            </td>
            <td>
              <a
                href="#role/users/9"
                onClick={[Function]}
              >
                Users
              </a>
                |  
              <a
                href="#role/9/add/user"
                onClick={[Function]}
              >
                Assign
              </a>
                |  
              <a
                href="#role/edit9"
                onClick={[Function]}
              >
                Edit
              </a>
                |  
              <a
                href="#role/delete9"
                onClick={[Function]}
              >
                Delete
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <a
      href="#role/add"
      onClick={[Function]}
    >
      Add Role
    </a>
  </div>
</div>
`;

exports[`Dashboard <Main /> renders User/Roles correctly 1`] = `
<div
  className="card"
  style={Object {}}
>
  <div
    className="card-header"
  >
    User &gt; User ID: 3 &gt; Roles
  </div>
  <div
    className="card-body"
  >
    <a
      href="#users"
      onClick={[Function]}
    >
      Back
    </a>
     
     
    <a
      href="#roles"
      onClick={[Function]}
    >
      Roles
    </a>
    <div
      className="table-responsive"
    >
      <table
        className="table table-sm table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              ID
            </th>
            <th>
              Name
            </th>
            <th>
              Options
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td
              style={Object {}}
            >
              Administrator
            </td>
            <td>
              <a
                href="#role/1/users"
                onClick={[Function]}
              >
                Other Users
              </a>
                |  
              <a
                href="#role/1/delete/user"
                onClick={[Function]}
              >
                Unassign Role
              </a>
            </td>
          </tr>
          <tr>
            <td>
              8
            </td>
            <td
              style={Object {}}
            >
              Everyone
            </td>
            <td>
              <a
                href="#role/8/users"
                onClick={[Function]}
              >
                Other Users
              </a>
                |  
              <a
                href="#role/8/delete/user"
                onClick={[Function]}
              >
                Unassign Role
              </a>
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td
              style={Object {}}
            >
              Myself
            </td>
            <td>
              <a
                href="#role/9/users"
                onClick={[Function]}
              >
                Other Users
              </a>
                |  
              <a
                href="#role/9/delete/user"
                onClick={[Function]}
              >
                Unassign Role
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <a
      href="#users"
      onClick={[Function]}
    >
      Back
    </a>
  </div>
</div>
`;

exports[`Dashboard <Main /> renders Users correctly 1`] = `
<div
  className="card"
  style={Object {}}
>
  <div
    className="card-header"
  >
    Users
  </div>
  <div
    className="card-body"
  >
    <div
      className="table-responsive"
    >
      <table
        className="table table-sm table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              ID
            </th>
            <th>
              Email
            </th>
            <th>
              Activated
            </th>
            <th>
              Options
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td>
              blah@blah.com
            </td>
            <td>
              Yes
            </td>
            <td>
              <a
                href="/user/profile"
                onClick={[Function]}
              >
                Profile
              </a>
                |  
              <a
                href="#user/1/roles"
                onClick={[Function]}
              >
                Roles
              </a>
            </td>
          </tr>
          <tr>
            <td>
              2
            </td>
            <td>
              blah2@blah.com
            </td>
            <td>
              Yes
            </td>
            <td>
              <a
                href="/user/profile"
                onClick={[Function]}
              >
                Profile
              </a>
                |  
              <a
                href="#user/2/roles"
                onClick={[Function]}
              >
                Roles
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
